sympy/core/expr.py

```diff
@@ -809,14 +809,16 @@ def _eval_interval(self, x, a, b):
                 if isinstance(B, Limit):
                     raise NotImplementedError("Could not compute limit")
 
+        if (a and b) is None:
+            return B - A
+
         value = B - A
-        a = a if a is not None else 0
-        b = b if b is not None else 0
-        singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
-        for s in singularities:
-            if all([S(a).is_number, S(s).is_number, S(b).is_number])\
-                    and a < s < b:
-                value += -limit(self, x, s, "+") + limit(self, x, s, "-")
+        if all([S(a).is_comparable, S(b).is_comparable]):
```

Comment:
`_eval_interval` is intended for internal use and it is expected that the caller has sympified its arguments. Actually, the test should be fixed: `assert f._eval_interval(x, S(-1), S(1)) == S(2)/5`